src/prefect/executors/dask.py

```diff
@@ -595,8 +593,7 @@ def _posttask(self, key, value, dsk, state, id):  # type: ignore
                     if exiting_early:
                         self._interrupt_pool()
                     else:
-                        self._pool.close()
-                    self._pool.join()
+                        self._pool.shutdown(wait=False)
```

Comment:
Shouldn't this be `wait=True` since it's not an early exit?

Reply:
I think that's right yes, fixed now.